packages/versioning/src/validate.ts

```diff
@@ -21,20 +21,17 @@ import {
 } from "./versioning.js";
 
 export function $onValidate(program: Program) {
-  const namespaceDependencies = new Map();
+  const namespaceDependencies = new Map<Namespace | undefined, Set<Namespace>>();
```

Comment:
This resulted in bugs because it loses the Typescript compiler's ability to catch issues. It would be good if we could use a linter rule to prohibit these kinds of patterns that let things get treated as `any` and just allow anything to happen